mqtt/mqtt.js

```diff
@@ -72,8 +72,14 @@ nats.subscribe('channel.*', function (msg) {
 
 aedes.authorizePublish = function (client, packet, publish) {
     // Topics are in the form `channels/<channel_id>/messages`
-    var channel = packet.topic.split('/')[1],
-        accessReq = {
+    var channel = /^channels\/(\d+)\/messages$/.exec(packet.topic);
+    if (channel == null) {
```

Comment:
In JS you usually check this stuff like `!channel`.

Reply:
done